text/chapter6.md

```diff
@@ -661,20 +662,32 @@ Another reason to define a superclass relationship is in the case where there is
        mempty = Multiply 1
      ```
 
-     This monoid acts on strings by repeating an input string some number of times. Write an instance which implements this action:
+     Write an instance which implements this action:
+
+     ```haskell
+     instance actionMultiplyInt :: Action Multiply Int
+     ```
+     Does this instance satisfy the laws listed above?
+
+1. (Medium) Write an `Action` instance which repeats an input string some number of times:
 
      ```haskell
-     instance repeatAction :: Action Multiply String
+     instance actionMultiplyString :: Action Multiply String
      ```
-     _Hint_: Search Pursuit for a helper-function with the signature `String -> Int -> String`. Note that `String` might appear as a more generic type.
+     _Hint_: Search Pursuit for a helper-function with the signature [`String -> Int -> String`](https://pursuit.purescript.org/search?q=String%20-%3E%20Int%20-%3E%20String). Note that `String` might appear as a more generic type (such as `Monoid`).
 
      Does this instance satisfy the laws listed above?
+
  1. (Medium) Write an instance `Action m a => Action m (Array a)`, where the action on arrays is defined by acting on each array element independently.
+
  1. (Difficult) Given the following newtype, write an instance for `Action m (Self m)`, where the monoid `m` acts on itself using `append`:
 
      ```haskell
      newtype Self m = Self m
      ```
+
+     _Note_: The testing framework requires `Show` and `Eq` instances for the `Self` and `Multiply` types. You may either write these instances manually, or let the compiler handle this for you with [`derive newtype instance`](https://github.com/purescript/documentation/blob/master/language/Type-Classes.md#type-class-deriving) shorthand.
```

Comment:
This link won't be much help until https://github.com/purescript/documentation/pull/338 is merged